cli/azd/pkg/alpha/alpha_feature.go

```diff
@@ -58,12 +70,21 @@ func GetEnableCommand(key FeatureId) string {
 	return fmt.Sprintf("azd config set %s on", strings.Join([]string{parentKey, string(key)}, "."))
 }
 
+// ShouldWarn returns true if a warning should be emitted when using a given alpha feature.
```

Comment:
Are there cases we think that `ShouldWarn()` would return `false`, but it's appropriate to perform `WarningMessage()` and output to console? Or can this be an implementation detail of `WarningMessage()`?

It seems that given the nature of the warning message, they should always be consistent.

Reply:
Sounds good - I'll fix this up.

Reply:
Refactored this such that there is now a WarnForFeature method on the console itself.  It manages the policy of if it should show the warning or not.